### DISCUSSION TITLE
Override .ckpt in ModelCheckpoint Callback

### DISCUSSION THREAD

**Discussion by victkid** [Lightning-AI/pytorch-lightning#5684] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi,

I'm using ModelCheckpoint Callback to save my model checkpoint. pytorch lightning automatically attaches `-v0`, `-v1` to the filename I specified if it finds checkpoint models exist in `dirpath`.  Instead of saving all the models from different runs, is there a way to make the ModelCheckpoint Callback only save one model in the checkpoint folder and just override the model from previous runs?

For example, my `ModelCheckpoint` is as follow: 
```
ModelCheckpoint(monitor='valid_score',
                              dirpath="./checkpoint/",
                              filename="model",
                              mode='max', save_top_k=1))
```
If I run the code for 3 three times, my `checkpoints` folder will have the following: 
```
- checkpoint:
    - model.ckpt
    - model-v0.ckpt
    - model-v1.ckpt
```
Would it be possible to just have `model.ckpt` in my `checkpoint` folder no matter how many times I run the code?

**Comment by carmocca** [comment, marked as answer]
We don't support this, but you could always remove the file manually between runs with `os.remove()`
